engine/src/main/java/org/kigalisim/engine/support/RecalcStrategy.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * Strategy interface for recalculation operations.
+ *
+ * <p>This interface defines the contract for recalculation strategies that can be
+ * executed on an Engine instance. Each strategy encapsulates a specific type of
+ * recalculation logic that was previously implemented as private methods in
+ * SingleThreadEngine.</p>
+ *
+ * @license BSD-3-Clause
+ */
+
+package org.kigalisim.engine.support;
+
+import org.kigalisim.engine.Engine;
+
+/**
+ * Strategy interface for recalculation operations.
+ */
+public interface RecalcStrategy {
+
+  /**
+   * Execute the recalculation strategy on the given engine.
+   *
+   * @param target The engine on which to execute the recalculation
+   */
+  void execute(Engine target);
```

Comment:
I'm worried about introducing too much for backwards compatibility so soon given the newness of this code. Can we place remove this and have all of the implementors take RecalcKit even if it is not needed? In other words, only have the execute form that has RecalcKit.